core/descriptors/src/org/jetbrains/kotlin/util/OperatorNameConventions.kt

```diff
@@ -34,6 +34,7 @@ object OperatorNameConventions {
     @JvmField val HAS_NEXT = Name.identifier("hasNext")
 
     @JvmField val COMPONENT_REGEX = Regex("component\\d+")
+    @JvmField val DECONSTRUCT = Name.identifier("deconstruct")
```

Comment:
Don't quite understand why this is needed. If you write a design document, please mention this moment there.